cpp/src/arrow/array.cc

```diff
@@ -1274,16 +1255,7 @@ struct ValidateVisitor {
     return Status::OK();
   }
 
-  Status Visit(const UnionArray& array) {
-    if (array.length() < 0) {
-      return Status::Invalid("Length was negative");
-    }
-
-    if (array.null_count() > array.length()) {
-      return Status::Invalid("Null count exceeds the length of this struct");
-    }
-    return Status::OK();
-  }
+  Status Visit(const UnionArray& array) { return Status::OK(); }
```

Comment:
Why did this validation get removed?

Reply:
Ah, it's not removed, it's factored out below into `ValidateArray`.